### ISSUE TITLE
Karabiner Elements not working on Monterey

### ISSUE THREAD

**Issue by MarcoSero** [pqrs-org/Karabiner-Elements#2967]
I have upgraded to Monterey and Karabiner Elements is no longer working. Everything seems to be running find but keys are simply not remapped at all. 

Mac OS Version: `12.1`
Karabiner Elements version: `14.3.0`

What I tried:

- [x] Completely uninstalled and reinstalled Karabiner Elements, approving the security alert on install
- [x] Tried both simple and complex modifications
- [x] Tried a different keyboard
- [x] Checked all permissions in Privacy and Security (see below)

<img width="663" alt="Screen Shot 2022-01-17 at 09 05 32" src="https://user-images.githubusercontent.com/867996/149731085-381276d2-db74-4dcb-b4b4-889f01d92b0d.png">
<img width="661" alt="Screen Shot 2022-01-17 at 09 05 39" src="https://user-images.githubusercontent.com/867996/149731095-b8541a8c-03e0-4556-b0e0-506f5e81bdd0.png">

I couldn't find anything useful/alarming in the logs either. Attached below

```
[2022-01-17 08:56:47.353] [info] [grabber] virtual_hid_device_service_client_ virtual_hid_keyboard_ready_response: true
[2022-01-17 08:56:47.372] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.378] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.379] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.387] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.390] [info] [grabber] caps lock is found on USB DEVICE
[2022-01-17 08:56:47.390] [info] [grabber] USB DEVICE (device_id:4294970539) is grabbed.
[2022-01-17 08:56:47.390] [info] [grabber] caps_lock_led_state_manager is started: USB DEVICE
[2022-01-17 08:56:47.391] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.391] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.393] [info] [grabber] caps lock is found on USB Receiver
[2022-01-17 08:56:47.393] [info] [grabber] USB DEVICE (device_id:4294970535) is grabbed.
[2022-01-17 08:56:47.394] [info] [grabber] caps_lock_led_state_manager is started: USB DEVICE
[2022-01-17 08:56:47.394] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.394] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:47.395] [info] [grabber] USB Receiver (device_id:4294969628) is grabbed.
[2022-01-17 08:56:47.395] [info] [grabber] caps_lock_led_state_manager is started: USB Receiver
[2022-01-17 08:56:47.395] [info] [virtual_hid_device_service] virtual_hid_device_service_clients_manager (pointing) client is removed (size: 0)
[2022-01-17 08:56:49.380] [info] [grabber] hid_event_system_monitor set_caps_lock_delay_override for pqrs.org:Karabiner-DriverKit-VirtualHIDKeyboard
```

**Comment by maciejkalis** [comment]
Do you have Intel or Silicon CPU?

**Comment by MarcoSero** [comment]
> Do you have Intel or Silicon CPU?

Intel - iMac Pro (2017)

**Comment by nicholasRenninger** [comment]
I am having the same issue for my Intel - 2019 MacBook Pro 16".

**Comment by Q-kil** [comment]
I am having the same issue for my M1 - Mac mini

**Comment by dnordberg** [comment]
Same 2018 MacBook Pro 15 Monterey 12.1,
Karabiner 14.3.0

**Comment by Alex-Just** [comment]
@dnordberg can confirm that it's caused by the mackup+dropbox integration. I was able to fix it by removing the file link, removing the `~/.config/karabiner` folder, and reinstalling karabiner. Not sure what was the exact issue but that did the trick

**Comment by lambainsaan** [comment]
> @dnordberg can confirm that it's caused by the mackup+dropbox integration. I was able to fix it by removing the file link, removing the `~/.config/karabiner` folder, and reinstalling karabiner. Not sure what was the exact issue but that did the trick

This worked, I had a link to the config folder that I had put in one drive, but after removing the link and restarting karabiner it started working properly. I had denied access to one drive when I updated my mac to Monetery, because I thought why would it need access to my drive, and looks like that caused the issue.

**Comment by mgiu96411** [comment]
Hey guys! Would love some assistance. I don't know what mackup is but I did download a Dropbox Beta + 12.3 Monterey update and my Karabiner complex modification stopped working.

I don't know programming so I tried what I could. I moved the .config folder to my desktop (to save the modification which was hard to find on the internet), deleted Karabiner and reinstalled. Then moved the folder back and tried doing a symlink (googling how):
`ln -s ~/.config/karabiner ~/.config
ln: /Users/giusse/.config/karabiner: File exists`

 So I tried removing it: 
`rm ~/.config/karabiner
rm: /Users/giusse/.config/karabiner: is a directory`

Then I tried this and different path variations
`rm /.config/karabiner
rm: /.config/karabiner: No such file or directory`.
I don't know what to do.

Btw, the complex modification is to map my mouse's side buttons to copy (button 5) and paste (button 4).
It makes my job 1000x easier and I'm having a very hard time now.

Thanks in advance!

**Comment by mgiu96411** [comment]
Hello again! I've realized that my issue isn't the symlink. A third part of the complex modification is that clicking the scroll wheel acts like cmd+shift+4, and that works. However, the side buttons on the mouse aren't being registered in Karabiner EventViewer. It's like my Mac can't see that they exist after the latest update. Any ideas?

**Comment by 0x3333** [comment]
My issue was that karabiner_grabber needed full disk access because I have my configuration folder symlinked.

**Comment by mgiu96411** [comment]
Hey! Thanks for the reply. I gave it full disk access already and that has not worked. Besides, the third part of the modification works. The problem is recognizing the two non-conventional side buttons.

**Comment by mycroftcanner** [comment]
```
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro17,1
  Chip:	Apple M1
  System Version:	macOS 12.3 (21E230)
  Kernel Version:	Darwin 21.4.0
```

```
3 extension(s)
--- com.apple.system_extension.driver_extension
enabled	active	teamID	bundleID (version)	name	[state]
*	*	G43BCU2T37	org.pqrs.Karabiner-DriverKit-VirtualHIDDevice (1.6.0/1.6.0)	org.pqrs.Karabiner-DriverKit-VirtualHIDDevice	[activated enabled]
```

Preferences are set correctly for karabiner's grabber, observer and event viewer.

The event viewer sees mouse 4 and mouse 5 on my Razer mouses.  

```
type:down            HID usage: 9,4       name:{"pointing_button":"button4"}                           misc:
type:up              HID usage: 9,4       name:{"pointing_button":"button4"}                           misc:
type:down            HID usage: 9,5       name:{"pointing_button":"button5"}                           misc:
type:up              HID usage: 9,5       name:{"pointing_button":"button5"}                           misc:
type:down            HID usage: 9,1       name:{"pointing_button":"button1"}                           misc:
type:up              HID usage: 9,1       name:{"pointing_button":"button1"}                           misc:
```

However modifications do not work. They work for Caplocks or anything else :(

**Comment by fk1blow** [comment]
Had karabiner installed since big sur, then upgraded to monterey and it kept working.
Transitioned to a new mac with monterey preinstalled - karabiner wouldn't work(no matter the permissions).
After openening the Karabiner event viewer and set the permissions(for it, specifically), karabiner elements started to work.

tl;dr open up karabiner event viewer, give the permissions (inside privacy, when and if asked), karabiner-elements may work!

**Comment by vienthuong** [comment]
Same issue on Monterey on M1 Pro

**Comment by Amatson** [comment]
Same issue with Ventura 13 M1 Pro

**Comment by willbchang** [comment]
> My issue was that karabiner_grabber needed full disk access because I have my configuration folder symlinked.

I also encountered dropbox symlink problem. This actually solved it.

<img width="488" alt="image" src="https://user-images.githubusercontent.com/14329786/211234723-517f7654-659d-4dff-94d2-4da2d68be444.png">
